packages/ragbits-document-search/src/ragbits/document_search/retrieval/rerankers/litellm.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+
+import litellm
+from pydantic import BaseModel
+
+from ragbits.document_search.documents.element import Element, TextElement
+from ragbits.document_search.retrieval.rerankers.base import Reranker
+
+
+class LiteLLMReranker(BaseModel, Reranker):
+    """
+    A LiteLLM reranker for providers such as Cohere, Together AI, Azure AI.
+    For more details and parameters see https://docs.litellm.ai/docs/rerank
+
+    Args:
+        model: The model to use.
+        top_n: The number of most relevant documents or indices to return.
+        return_documents: Whether to return the documents or just indices and relevalnce scores.
+        rank_fields: The fields to rank on.
+        max_chunks_per_doc: The maximum number of chunks per document to consider.
+
+    Returns:
+        The reranked chunks.
+    """
+
+    model: str
+    top_n: int | None = None
+    return_documents: bool = False
+    rank_fields: list[str] | None = None
+    max_chunks_per_doc: int | None = None
+
+    async def rerank(self, chunks: List[Element], query: str) -> List[Element]:
+        """
+        Reranking with LiteLLM API.
+
+        Args:
+            chunks: The chunks to rerank.
+            query: The query to rerank the chunks against.
+
+        Returns:
+            The reranked chunks.
+
+        Raises:
+            ValueError: If chunks are not a list of TextElement objects.
+        """
+        if not all(isinstance(chunk, TextElement) for chunk in chunks):
```

Comment:
Why only `TextElements` here?

Reply:
I think that is partially because of that how `Element` base model is designed. What we're lacking is some sort of abstract method that would return text representation of the element. (e.g. `.to_text()`). Then we could use that unified interface while passing elements to LLMs, rerankers, etc.

Reply:
Created #143